packages/loader/container-loader/src/container.ts

```diff
@@ -880,20 +880,20 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
 
             // post summary here
             if (hasAttachmentBlobs) {
-                const appSummary: ISummaryTree = this.context.createSummary();
+                const appSummary: ISummaryTree = this.context.createSummary(new Map<string, string>());
                 const protocolSummary = this.captureProtocolSummary();
                 summary = combineAppAndProtocolSummary(appSummary, protocolSummary);
 
                 this._attachState = AttachState.Attaching;
                 this.context.notifyAttaching();
 
-                await this.storageService.uploadSummaryWithContext(summary, {
+                await (this.storageService as any).uploadSummaryWithContext(summary, {
                     referenceSequenceNumber: 0,
                     ackHandle: undefined,
                     proposalHandle: undefined,
-                });
+                }, true);
```

Comment:
Let's move the flag to context, but if it's its own argument, would love to see comments on what it means, i.e.

true /* initial summary */)